### DISCUSSION TITLE
Install OFW Patches

### DISCUSSION THREAD

**Discussion by richi1993** [toniebox-reverse-engineering/hackiebox_cfw_ng#39] [General]
Hello together,
i want to install some of the OFW-Patches but I have some problems.

First I need to find out, if the Patch is compatible or not. 
The only way I can do this, is to connect the Chip with PC and use the firmware_info tool?

And when I know the Version, what should I do with the JSON-Patches?
How can I install/use them?

Thank you for your work!

**Comment by SciLor** [comment]
Hey @richi1993,
first of all you should create a backup of your toniebox' flash. Then check which firmware version you have available. the 3.1.0_BF2 would be the one recommended, as most of the patches work there.
If you have installed the custom booloader, you'll just need to place the firmware on your sd in the right location as ng-ofw3 and boot it.
